### PR TITLE
ESI-11996 Protect against migrations race optional

### PR DESCRIPTION
**Pull request by jkmar**
#488 
#508 

Now usage of etcd during migrations is optional and requires a flag.

### CODE REVIEW

**Comment by jkmar** [comment]
@morrisson 
Please review.

File: sync/util/etcd.go [hunk 1]
```diff
@@ -14,24 +14,23 @@ import (
...
 
 // CreateFromConfig creates etcd sync from config
-func CreateFromConfig(config *util.Config) (s sync.Sync, err error) {
+func CreateFromConfig(config *util.Config) (etcdSync sync.Sync, err error) {
 	syncType := config.GetString("sync", "etcd")
+	etcdServers := config.GetStringList("etcd", nil)
+	if etcdServers == nil {
+		err = fmt.Errorf("no etcd found in config file")
+		return
+	}
```
Comment (morrisson) [inline sync/util/etcd.go]: Basically, gohan can work without etcd. These lines force gohan to work with etcd.

File: cli/cli.go [hunk 2]
```diff
@@ -391,23 +395,53 @@ documentation for detail information about writing tests.`,
...
+func migrationSubcommandWithLock(action func(*cli.Context)) func(*cli.Context) {
+	return func(context *cli.Context) {
+		configFile := context.String("config-file")
+		if migration.LoadConfig(configFile) != nil {
+			return
+		}
+
+		useEtcd := context.Bool(useEtcdDuringMigrationsFlag)
+		if !useEtcd {
+			log.Warning("Do not run multiple processes in parallel")
```
Comment (morrisson) [inline cli/cli.go]: I think this message make people misunderstand that they have already executed another migration process.

**Comment by jkmar** [comment]
@morrisson 
I've reverted changes that i've made to sync/util/etcd.go (it returns nil sync instead of error).
Please review again.

**Comment by jkmar** [comment]
@marcin-ptaszynski 
Please review.

File: cli/cli.go [hunk 3]
```diff
@@ -394,23 +398,52 @@ documentation for detail information about writing tests.`,
...
+	}
+}
+
 func getMigrateSubcommand(subcmd, usage string) cli.Command {
 	return cli.Command{
 		Name:  subcmd,
 		Usage: usage,
 		Flags: []cli.Flag{
 			cli.StringFlag{Name: "config-file", Value: defaultConfigFile, Usage: "Server config File"},
+			cli.BoolFlag{Name: useEtcdDuringMigrationsFlag, Usage: "Enable if ETCD should be used to synchronize migrations"},
```
Comment (marcin-ptaszynski) [inline cli/cli.go]: lockWithEtcd?

**Comment by jkmar** [comment]
@nati 
Please review.

**Pull request merged**